accounts/abi/abi.go

```diff
@@ -117,11 +118,80 @@ func (abi ABI) Pack(name string, args ...interface{}) ([]byte, error) {
 	return append(method.Id(), arguments...), nil
 }
 
+// toGoSliceType prses the input and casts it to the proper slice defined by the ABI
+// argument in T.
+func toGoSlice(i int, t Argument, output []byte) (interface{}, error) {
+	index := i * 32
+	// The slice must, at very least be large enough for the index+32 which is exactly the size required
+	// for the [offset in output, size of array]. The size of the array (in elements) is exactly 32 bytes.
```

Comment:
This check can only ensure that the "offse" value fits in, it cannot validate the actual offset or size without parsing it.